File: java2csharp/java/Microsoft.Mobile.Engagement/engagement/EngagementConfiguration.java
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

package com.microsoft.azure.engagement;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Engagement configuration. This contains configuration that can be dynamically changed at runtime.
 * All the fields are optional.
 */
public class EngagementConfiguration implements Parcelable
{
  /** Prefix for keys */
  private static final String PREFIX = "engagement:";

  /** Location report key prefix */
  private static final String LOCATION_REPORT_PREFIX = PREFIX + "locationReport:";

  /** Agent enabled setting */
  public static final String ENABLED = PREFIX + "enabled";

  /** Lazy location report flag */
  public static final String LOCATION_REPORT_LAZY_AREA = LOCATION_REPORT_PREFIX + "lazyArea";

  /** Realtime location report flag */
  public static final String LOCATION_REPORT_REAL_TIME = LOCATION_REPORT_PREFIX + "realTime";

  /** Realtime location report flag (fine mode) */
  public static final String LOCATION_REPORT_REAL_TIME_FINE = LOCATION_REPORT_REAL_TIME + ":fine";

  /** Realtime location report flag (background mode) */
  public static final String LOCATION_REPORT_REAL_TIME_BACKGROUND = LOCATION_REPORT_REAL_TIME
    + ":background";

  /** Parcelable factory */
  public static final Parcelable.Creator<EngagementConfiguration> CREATOR = new Parcelable.Creator<EngagementConfiguration>()
  {
    @Override
    public EngagementConfiguration createFromParcel(Parcel in)
    {
      return new EngagementConfiguration(in);
    }

    @Override
    public EngagementConfiguration[] newArray(int size)
    {
      return new EngagementConfiguration[size];
    }
  };

  /** True serialized in Parcelable */
  private static final byte TRUE = Byte.MAX_VALUE;

  /** False serialized in Parcelable */
  private static final byte FALSE = 0;

  /**
   * Serialize a boolean for Parcelable.
   * @param value boolean to serialize.
   * @return byte value to use in Parcelable.
   */
  private static byte toByte(boolean value)
  {
    return value ? TRUE : FALSE;
  }

  /**
   * Parse a boolean from Parcelable.
   * @param value value to parse from Parcelable.
   * @return parsed boolean.
   */
  private static boolean toBoolean(byte value)
  {
    return value == TRUE;
  }

  /** Connection string */
  private String mConnectionString;

  /** Lazy area location report flag */
  private boolean mLazyAreaLocationReport;

  /** Real time location report flag */
  private boolean mRealtimeLocationReport;

  /** Fine real time location report flag */
  private boolean mFineRealtimeLocationReport;

  /** Background real time location report flag */
  private boolean mBackgroundRealtimeLocationReport;

  /** Init. */
  public EngagementConfiguration()
  {
  }

  /**
   * Unmarshal a parcel.
   * @param in parcel.
   */
  private EngagementConfiguration(Parcel in)
  {
    mConnectionString = in.readString();
    mLazyAreaLocationReport = toBoolean(in.readByte());
    mRealtimeLocationReport = toBoolean(in.readByte());
    mFineRealtimeLocationReport = toBoolean(in.readByte());
    mBackgroundRealtimeLocationReport = toBoolean(in.readByte());
  }

  @Override
  public void writeToParcel(Parcel dest, int flags)
  {
    dest.writeString(mConnectionString);
    dest.writeByte(toByte(mLazyAreaLocationReport));
    dest.writeByte(toByte(mRealtimeLocationReport));
    dest.writeByte(toByte(mFineRealtimeLocationReport));
    dest.writeByte(toByte(mBackgroundRealtimeLocationReport));
  }

  @Override
  public int describeContents()
  {
    return 0;
  }

  /**
   * Get connection string.
   * @return connection string.
   */
  public String getConnectionString()
  {
    return mConnectionString;
  }

  /**
   * Set connection string.
   * @param connectionString connection string.
   */
  public void setConnectionString(String connectionString)
  {
    mConnectionString = connectionString;
  }

  /**
   * Get lazy area location report flag.
   * @return lazy area location report flag.
   */
  public boolean isLazyAreaLocationReport()
  {
    return mLazyAreaLocationReport;
  }

  /**
   * Set lazy area location report flag.
   * @param lazyAreaLocationReport lazy area location report flag.
   */
  public void setLazyAreaLocationReport(boolean lazyAreaLocationReport)
  {
    mLazyAreaLocationReport = lazyAreaLocationReport;
  }

  /**
   * Get real time location report flag.
   * @return real time location report flag.
   */
  public boolean isRealtimeLocationReport()
  {
    return mRealtimeLocationReport;
  }

  /**
   * Set real time location report flag.
   * @param realtimeLocationReport real time location report flag.
   */
  public void setRealtimeLocationReport(boolean realtimeLocationReport)
  {
    mRealtimeLocationReport = realtimeLocationReport;
  }

  /**
   * Get fine real time location report flag.
   * @return fine real time location report flag.
   */
  public boolean isFineRealtimeLocationReport()
  {
    return mFineRealtimeLocationReport;
  }

  /**
   * Set fine real time location report flag.
   * @param fineRealtimeLocationReport fine real time location report flag.
   */
  public void setFineRealtimeLocationReport(boolean fineRealtimeLocationReport)
  {
    mFineRealtimeLocationReport = fineRealtimeLocationReport;
  }

  /**
   * Get background real time location report flag.
   * @return background real time location report flag.
   */
  public boolean isBackgroundRealtimeLocationReport()
  {
    return mBackgroundRealtimeLocationReport;
  }

  /**
   * Set background real time location report flag.
   * @param backgroundRealtimeLocationReport background real time location report flag.
   */
  public void setBackgroundRealtimeLocationReport(boolean backgroundRealtimeLocationReport)
  {
    mBackgroundRealtimeLocationReport = backgroundRealtimeLocationReport;
  }
}

File: java2csharp/csharp/Microsoft.Mobile.Engagement/engagement/EngagementConfiguration.cs
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 */

namespace com.microsoft.azure.engagement
{

	using Parcel = android.os.Parcel;
	using Parcelable = android.os.Parcelable;

	/// <summary>
	/// Engagement configuration. This contains configuration that can be dynamically changed at runtime.
	/// All the fields are optional.
	/// </summary>
	public class EngagementConfiguration : Parcelable
	{
	  /// <summary>
	  /// Prefix for keys </summary>
	  private const string PREFIX = "engagement:";

	  /// <summary>
	  /// Location report key prefix </summary>
	  private static readonly string LOCATION_REPORT_PREFIX = PREFIX + "locationReport:";

	  /// <summary>
	  /// Agent enabled setting </summary>
	  public static readonly string ENABLED = PREFIX + "enabled";

	  /// <summary>
	  /// Lazy location report flag </summary>
	  public static readonly string LOCATION_REPORT_LAZY_AREA = LOCATION_REPORT_PREFIX + "lazyArea";

	  /// <summary>
	  /// Realtime location report flag </summary>
	  public static readonly string LOCATION_REPORT_REAL_TIME = LOCATION_REPORT_PREFIX + "realTime";

	  /// <summary>
	  /// Realtime location report flag (fine mode) </summary>
	  public static readonly string LOCATION_REPORT_REAL_TIME_FINE = LOCATION_REPORT_REAL_TIME + ":fine";

	  /// <summary>
	  /// Realtime location report flag (background mode) </summary>
	  public static readonly string LOCATION_REPORT_REAL_TIME_BACKGROUND = LOCATION_REPORT_REAL_TIME + ":background";

	  /// <summary>
	  /// Parcelable factory </summary>
	  public static readonly Parcelable.Creator<EngagementConfiguration> CREATOR = new CreatorAnonymousInnerClassHelper();

	  private class CreatorAnonymousInnerClassHelper : Parcelable.Creator<EngagementConfiguration>
	  {
		  public CreatorAnonymousInnerClassHelper()
		  {
		  }

		  public override EngagementConfiguration createFromParcel(Parcel @in)
		  {
			return new EngagementConfiguration(@in);
		  }

		  public override EngagementConfiguration[] newArray(int size)
		  {
			return new EngagementConfiguration[size];
		  }
	  }

	  /// <summary>
	  /// True serialized in Parcelable </summary>
	  private static readonly sbyte TRUE = SByte.MaxValue;

	  /// <summary>
	  /// False serialized in Parcelable </summary>
	  private const sbyte FALSE = 0;

	  /// <summary>
	  /// Serialize a boolean for Parcelable. </summary>
	  /// <param name="value"> boolean to serialize. </param>
	  /// <returns> byte value to use in Parcelable. </returns>
	  private static sbyte toByte(bool value)
	  {
		return value ? TRUE : FALSE;
	  }

	  /// <summary>
	  /// Parse a boolean from Parcelable. </summary>
	  /// <param name="value"> value to parse from Parcelable. </param>
	  /// <returns> parsed boolean. </returns>
	  private static bool toBoolean(sbyte value)
	  {
		return value == TRUE;
	  }

	  /// <summary>
	  /// Connection string </summary>
	  private string mConnectionString;

	  /// <summary>
	  /// Lazy area location report flag </summary>
	  private bool mLazyAreaLocationReport;

	  /// <summary>
	  /// Real time location report flag </summary>
	  private bool mRealtimeLocationReport;

	  /// <summary>
	  /// Fine real time location report flag </summary>
	  private bool mFineRealtimeLocationReport;

	  /// <summary>
	  /// Background real time location report flag </summary>
	  private bool mBackgroundRealtimeLocationReport;

	  /// <summary>
	  /// Init. </summary>
	  public EngagementConfiguration()
	  {
	  }

	  /// <summary>
	  /// Unmarshal a parcel. </summary>
	  /// <param name="in"> parcel. </param>
	  private EngagementConfiguration(Parcel @in)
	  {
		mConnectionString = @in.readString();
		mLazyAreaLocationReport = toBoolean(@in.readByte());
		mRealtimeLocationReport = toBoolean(@in.readByte());
		mFineRealtimeLocationReport = toBoolean(@in.readByte());
		mBackgroundRealtimeLocationReport = toBoolean(@in.readByte());
	  }

	  public override void writeToParcel(Parcel dest, int flags)
	  {
		dest.writeString(mConnectionString);
		dest.writeByte(toByte(mLazyAreaLocationReport));
		dest.writeByte(toByte(mRealtimeLocationReport));
		dest.writeByte(toByte(mFineRealtimeLocationReport));
		dest.writeByte(toByte(mBackgroundRealtimeLocationReport));
	  }

	  public override int describeContents()
	  {
		return 0;
	  }

	  /// <summary>
	  /// Get connection string. </summary>
	  /// <returns> connection string. </returns>
	  public virtual string ConnectionString
	  {
		  get
		  {
			return mConnectionString;
		  }
		  set
		  {
			mConnectionString = value;
		  }
	  }


	  /// <summary>
	  /// Get lazy area location report flag. </summary>
	  /// <returns> lazy area location report flag. </returns>
	  public virtual bool LazyAreaLocationReport
	  {
		  get
		  {
			return mLazyAreaLocationReport;
		  }
		  set
		  {
			mLazyAreaLocationReport = value;
		  }
	  }


	  /// <summary>
	  /// Get real time location report flag. </summary>
	  /// <returns> real time location report flag. </returns>
	  public virtual bool RealtimeLocationReport
	  {
		  get
		  {
			return mRealtimeLocationReport;
		  }
		  set
		  {
			mRealtimeLocationReport = value;
		  }
	  }


	  /// <summary>
	  /// Get fine real time location report flag. </summary>
	  /// <returns> fine real time location report flag. </returns>
	  public virtual bool FineRealtimeLocationReport
	  {
		  get
		  {
			return mFineRealtimeLocationReport;
		  }
		  set
		  {
			mFineRealtimeLocationReport = value;
		  }
	  }


	  /// <summary>
	  /// Get background real time location report flag. </summary>
	  /// <returns> background real time location report flag. </returns>
	  public virtual bool BackgroundRealtimeLocationReport
	  {
		  get
		  {
			return mBackgroundRealtimeLocationReport;
		  }
		  set
		  {
			mBackgroundRealtimeLocationReport = value;
		  }
	  }

	}

}